_guides/managing-oauth-clients.md

```diff
@@ -0,0 +1,209 @@
+     }
+   }'
+```
+
+#### To retrieve OAuth clients in a tenant, we will use the following API request:
+
+`GET https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients`
+
+Below are request payload parameters:
+
+| **Parameter**       | **Required** | **Description**        |
+|---------------------|--------------|------------------------|
+| `TENANT_ID`         | yes          | The ID of the tenant   |
+| `filter[component]` | no           | Filter by component ID |
+
+**EXAMPLES:**
+
+The first example is without filtering by component ID.
+
+`curl https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients \    -u
+{EMAIL}:{APIKEY}`
+
+The second example includes filtering by component ID.
+
+`curl
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/?filter[component]={{COMPONENT_ID}}
+\    -u {EMAIL}:{APIKEY}`
+
+#### OAuth clients can also be retrieved by their own ID, using the following API request:
+
+`curl
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID} \
+-u {EMAIL}:{APIKEY}`
+
+Below are request payload parameters:
+
+| **Parameter**     | **Required** | **Description**            |
+|-------------------|--------------|----------------------------|
+| `TENANT_ID`       | yes          | The ID of the tenant       |
+| `OAUTH-CLIENT_ID` | yes          | The ID of the OAuth client |
+
+**EXAMPLE:**
+
+`curl
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID} \
+-u {EMAIL}:{APIKEY}`
+
+#### To update OAuth clients in a tenant, we will use the following API request:
+
+`PATCH
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID}`
+
+Below are request payload parameters:
+
+| **Parameter**                       | **Required** | **Description**                    |
+|-------------------------------------|--------------|------------------------------------|
+| `type`                              | yes          | The value should be `oauth-client` |
+| `attributes.client_id`              | yes          | Oauth-client ID                    |
+| `attributes.client_secret`          | yes          | Oauth-client secret                |
+| `relationships.component.data.id`   | yes          | Component ID                       |
+| `relationships.component.data.type` | yes          | The value should be `component`    |
+
+**EXAMPLE:**
+
+```
+curl https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID} \
+  -X PATCH \
+  -u {EMAIL}:{APIKEY} \
+  -H 'Content-Type: application/json' -d '
+  {  
+    "data":{  
+      "type":"oauth-client",
+      "attributes":{  
+        "client_id":"{CLIENT_ID}",
+        "client_secret":"{CLIENT_SECRET}"
+      },
+      "relationships":{  
+        "component":{  
+          "data":{  
+            "id":"{COMPONENT_ID}",
+            "type":"component"
+          }
+        }
+      }
+    }
+  }'
+  ```
+
+#### To delete an OAuth client, we will use the following API request:
+
+`DELETE
```

Comment:
Notice: it's not possible to delete oauth client, if it is used. 
(oauth client is called used, if there exists flow in tenant that uses component, for which oauth client is defined)

Reply:
Done